django/cantusdb_project/main_app/views/chant.py

```diff
@@ -813,19 +829,36 @@ def get_queryset(self) -> QuerySet:
             # Filter the QuerySet with Q object
             chant_set = chant_set.filter(q_obj_filter)
             sequence_set = sequence_set.filter(q_obj_filter)
+            # Fetch only the values necessary for rendering the template
+            chant_set = chant_set.values(*values_for_rendering_template)
+            sequence_set = sequence_set.values(*values_for_rendering_template)
             # Finally, do keyword searching over the querySet
             if self.request.GET.get("keyword"):
                 keyword = self.request.GET.get("keyword")
-                # the operation parameter can be "contains" or "starts_with"
-                if self.request.GET.get("op") == "contains":
-                    chant_set = keyword_search(chant_set, keyword)
-                    sequence_set = keyword_search(sequence_set, keyword)
+                operation = self.request.GET.get("op")
+                if operation == "contains":
+                    ms_spelling_filter = Q(
+                        manuscript_full_text__icontains=keyword
+                    )
+                    std_spelling_filter = Q(
+                        manuscript_full_text_std_spelling__icontains=keyword
+                    )
+                    keyword_filter = ms_spelling_filter | std_spelling_filter
+                    chant_set = chant_set.filter(keyword_filter)
+                    sequence_set = sequence_set.filter(keyword_filter)
                 else:
-                    chant_set = chant_set.filter(incipit__istartswith=keyword)
-                    sequence_set = sequence_set.filter(incipit__istartswith=keyword)
+                    ms_spelling_filter = Q(
```

Comment:
There's a lot of duplicate code between the two cases of the if statement.

Reply:
good point - thank you!